sbi/utils/user_input_checks_utils.py

```diff
@@ -319,6 +345,16 @@ def support(self):
             reinterpreted_batch_ndims=1,
         )
 
+    def to(self, device):
+        # Move the values of the arg_constraints dictionary to the specified device
+        for i in range(len(self.dists)):
+            if hasattr(self.dists[i], "to"):
+                self.dists[i].to(device)
```

Comment:
`self.dists[i].to(device)  # type: ignore`

But please leave a brief comment about why this is needed :)